bots/guess_bot.js

```diff
@@ -0,0 +1,65 @@
+require('dotenv').config({silent: true})
+
+var Connection = require('../connection')
+
+const startWord = 'start'
+const guessWord = 'guess'
+const beginWord = 'begin'
+const rewardWord = 'reward'
+
+const startRE = new RegExp(startWord)
+const guessRE = new RegExp(guessWord)
+const beginRE = new RegExp(beginWord)
+const rewardRE = new RegExp(rewardWord)
+
+class GuessBot {
+  constructor() {
+    this.SLACK_CHANNEL = process.env.SLACK_CHANNEL
+    this.connection = new Connection()
+    this.connection.listen({
+      channel: this.SLACK_CHANNEL,
+      callback: this._handleMessage.bind(this),
+    })
+    this.connection.start().then(this._setId)
+  }
+
+  _setId = (id) => {
+    this.id = id
+    this.myselfRE = new RegExp(id)
+  }
+
+  _handleBegin(text, godBotId) {
+    const parts = text.split(beginRE)
+    const actionSpace = parts[1].trim().split(', ')
+    this.handleExplorationSpace(actionSpace, godBotId)
+  }
+
+  _handleReward(text) {
+    const parts = text.split(rewardRE)
+    const [reward, guess, totalScore, remaining] = parts[1].trim().split(', ')
+    this.handleReward(reward, guess, totalScore, remaining)
+  }
+
+  _handleMessage(message) {
+    const { text } = message
+    if (message.subtype === 'bot_message') {
```

Comment:
I was a little lazy with this, so I'm glad you looked into it. The case this is handling is when Slackbot posts the message "You mentioned someone, but they don't seem to be in the channel. Would you like to invite them?". That seemed like a common enough issue for people to hit when testing their bot in a channel, so I added a message. The tone of the message was intentionally uncertain because I anticipated there may be other "bot messages".

What is curious to me though, now that you point it out, is why every message from our bots aren't showing up as a `bot_message`. But for whatever reason, they are not.

This is what this specific message looks like to a human user, which to be honest, I've never seen anything like before (this must be relatively new)
<img width="621" alt="Screen Shot 2020-07-20 at 23 16 13" src="https://user-images.githubusercontent.com/3329425/88008899-3db4d380-cadf-11ea-8e00-9516ebb03038.png">

This is what the message looks like to a bot:
```
{
  type: 'message',
  subtype: 'bot_message',
  text: 'You mentioned <@U017GNL24JY>, but they’re not in this private channel.',
  is_ephemeral: true,
  user: 'USLACKBOT',
  username: 'slackbot',
  bot_id: 'B01',
  attachments: [
    {
      callback_id: 'consistentephemeralmentions_U017GNL24JY_1595301567001200_0',
      fallback: 'You may want to invite them.',
      id: 1,
      actions: [Array],
      mrkdwn_in: [Array]
    }
  ],
  channel: 'G016L0CBRT7',
  ts: '1595301567.001201',
  event_ts: '1595301568.000100'
}
```

I believe I could make this better by testing for `is_ephemeral` either in addition, or instead of testing `subtype`. What do you think?